cmd/zoekt-sourcegraph-indexserver/cleanup.go

```diff
@@ -515,3 +549,68 @@ func removeTombstones(fn string) ([]*zoekt.Repository, error) {
 	}
 	return tombstones, nil
 }
+
+// deleteShards deletes all the shards in indexDir that are associated with
+// the given repoID. If the repository specified by repoID happens to be in a
+// compound shard, the repository is tombstoned instead.
+//
+// deleteShards returns errRepositoryNotFound if the repository specified by repoID
+// isn't present in indexDir.
+//
+// Users must hold the global indexDir lock before calling deleteShards.
+func deleteShards(indexDir string, repoID uint32) error {
```

Comment:
We have this already in cleanup:

```
	simple := shards[:0]
		for _, s := range shards {
			if shardMerging && maybeSetTombstone([]shard{s}, repo) {
				shardsLog(indexDir, "tombname", []shard{s})
			} else {
				simple = append(simple, s)
			}
		}

		if len(simple) == 0 {
			continue
		}

		removeAll(simple...)
```

Can't we just call that instead? I guess that is what @keegancsmith was refering to. We could factor it out into its own function and call it from your handler and from cleanup. WDYT?